File: packages/csharp/src/Anip.Core/FinancialCost.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class FinancialCost
{
    [JsonPropertyName("currency")]
    public string Currency { get; set; } = "";

    [JsonPropertyName("amount")]
    public double? Amount { get; set; }

    [JsonPropertyName("range_min")]
    public double? RangeMin { get; set; }

    [JsonPropertyName("range_max")]
    public double? RangeMax { get; set; }

    [JsonPropertyName("typical")]
    public double? Typical { get; set; }

    [JsonPropertyName("upper_bound")]
    public double? UpperBound { get; set; }
}

File: packages/java/anip-core/src/main/java/dev/anip/core/FinancialCost.java
package dev.anip.core;

/**
 * Describes the financial cost of a capability invocation.
 */
public class FinancialCost {

    private String currency;
    private Double amount;
    private Double rangeMin;
    private Double rangeMax;
    private Double typical;
    private Double upperBound;

    /** No-arg constructor for Jackson deserialization. */
    public FinancialCost() {}

    public FinancialCost(String currency, Double amount, Double rangeMin,
                         Double rangeMax, Double typical, Double upperBound) {
        this.currency = currency;
        this.amount = amount;
        this.rangeMin = rangeMin;
        this.rangeMax = rangeMax;
        this.typical = typical;
        this.upperBound = upperBound;
    }

    /** Convenience constructor for fixed costs. */
    public FinancialCost(String currency, Double amount) {
        this(currency, amount, null, null, null, null);
    }

    public String getCurrency() {
        return currency;
    }

    public Double getAmount() {
        return amount;
    }

    public Double getRangeMin() {
        return rangeMin;
    }

    public Double getRangeMax() {
        return rangeMax;
    }

    public Double getTypical() {
        return typical;
    }

    public Double getUpperBound() {
        return upperBound;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    public void setAmount(Double amount) {
        this.amount = amount;
    }

    public void setRangeMin(Double rangeMin) {
        this.rangeMin = rangeMin;
    }

    public void setRangeMax(Double rangeMax) {
        this.rangeMax = rangeMax;
    }

    public void setTypical(Double typical) {
        this.typical = typical;
    }

    public void setUpperBound(Double upperBound) {
        this.upperBound = upperBound;
    }
}
